src/Locations.js

```diff
@@ -0,0 +1,58 @@
+import React, { Component } from "react";
+import axios from "axios";
+
+export class Locations extends Component {
+  constructor() {
+    super();
+    this.state = {
+      buttonText: "Show Locations",
+      locations: [],
+    };
+  }
+
+  handleClick = async () => {
+    let { data } = await axios.get("https://ghibliapi.herokuapp.com/locations");
+    data.forEach((elem) => {
+      if (elem.climate === "TODO") {
+        elem.climate = "Unknown";
+      }
+      if (elem.terrain === "TODO") {
+        elem.terrain = "Unknown";
+      }
+    });
+    if (this.state.buttonText === "Show Locations") {
+      this.setState({
+        locations: data,
+        buttonText: "Hide Locations",
+      });
+    } else {
+      this.setState({
+        locations: [],
+        buttonText: "Show Locations",
+      });
+    }
+  };
+  render() {
+    const { buttonText, locations } = this.state;
+    const renderLocations = () => {
```

Comment:
renderLocations doesn't have to be a function. You can also write it as:

```
const renderLocations = locations.map((elem, i) => {
    return (
      <li key={i}>
        <div>Name: {elem.name}</div>
        <div>Climate: {elem.climate}</div>
        <div>Terrain: {elem.terrain}</div>
      </li>
    );
});
```
Then on line 52, you can remove the parentheses